source/Calamari/Kubernetes/ResourceStatus/ResourceStatusChecker.cs

```diff
@@ -112,7 +192,18 @@ private static DeploymentStatus GetDeploymentStatus(List<Resource> resources, Li
             return DeploymentStatus.InProgress;
         }
 
-        private void LogInProgressResources(List<Resource> definedResourceStatuses, Dictionary<string, Resource> resourceStatuses, List<ResourceIdentifier> definedResources)
+        private void AddResourcesInternal(IEnumerable<ResourceIdentifier> newResources)
+        {
+            foreach (var resource in newResources)
+            {
+                lock (resourceLock)
+                {
+                    resources.Add(resource);
```

Comment:
There is no `AddRange` extension method for `HashSet`.. at least that I could see. 😢